src/services/fileStorage/proxy-service.js

```diff
@@ -455,13 +497,19 @@ const directoryService = {
 			sendPermissions = teamObject ? teamObject.filePermission : [];
 		}
 
-		const props = sanitizeObj(Object.assign(data, {
-			isDirectory: true,
-			owner: owner || userId,
-			parent,
-			refOwnerModel: owner ? (isCourse ? 'course' : 'teams') : 'user',
-			permissions: [...permissions, ...sendPermissions].map(setRefId),
-		}));
+		const props = sanitizeObj(
+			Object.assign(data, {
+				isDirectory: true,
+				owner: owner || userId,
+				parent,
+				refOwnerModel: () => {
```

Comment:
`Object.assign` does not copy functions. You can just use `refOwnerModel` because you already declared it.

Reply:
fixed